raiden/storage/wal.py

```diff
@@ -66,19 +59,20 @@ def restore_to_state_change(
 ST = TypeVar("ST", bound=State)
 
 
-class SavedState(NamedTuple):
+@dataclass(frozen=True)
```

Comment:
Can the `Generic[ST]` also be added to a `NamedTuple`? I generally don't know what's the point of changing something that has no functions to a frozen data class when a `NamedTuple` is enough.

Reply:
> Can the Generic[ST] also be added to a NamedTuple?

No, it will give an error.